go/vt/vtgate/planbuilder/operators/aggregator.go

```diff
@@ -117,13 +115,39 @@ func (a *Aggregator) isDerived() bool {
 	return a.TableID != nil
 }
 
+func (a *Aggregator) findCol(ctx *plancontext.PlanningContext, expr sqlparser.Expr) (int, error) {
+	if a.isDerived() {
+		derivedTBL, err := ctx.SemTable.TableInfoFor(*a.TableID)
+		if err != nil {
+			return 0, err
+		}
+		expr = semantics.RewriteDerivedTableExpression(expr, derivedTBL)
+	}
+	if offset, found := canReuseColumn(ctx, a.Columns, expr, extractExpr); found {
+		return offset, nil
+	}
+	return -1, nil
+}
+
 func (a *Aggregator) AddColumn(ctx *plancontext.PlanningContext, expr *sqlparser.AliasedExpr, _, addToGroupBy bool) (ops.Operator, int, error) {
-	if addToGroupBy {
-		return nil, 0, vterrors.VT13001("did not expect to add group by here")
+	offset, err := a.findCol(ctx, expr.Expr)
+	if err != nil {
+		return nil, 0, err
+	}
+	if offset >= 0 {
+		return a, offset, nil
+	}
+	if a.isDerived() {
+		derivedTBL, err := ctx.SemTable.TableInfoFor(*a.TableID)
+		if err != nil {
+			return nil, 0, err
+		}
+		expr.Expr = semantics.RewriteDerivedTableExpression(expr.Expr, derivedTBL)
 	}
+
 	// Aggregator is little special and cannot work if the input offset are not matched with the aggregation columns.
 	// So, before pushing anything from above the aggregator offset planning needs to be completed.
-	err := a.planOffsets(ctx)
```

Comment:
the `addToGroupBy` is moved down. Still, it should be done just after checking for `a.findCol < 0`